rfcs/006 - Deleting blocked resources asynchronously.md

```diff
@@ -0,0 +1,82 @@
+# 006 - Deleting blocked resources asynchronously
+
+## Current Status
+
+### Proposed
+
+2022-11-22
+
+### Accepted
+
+2022-11-11
+
+#### Approvers
+
+- Luis Fernando De Pombo <luisfer@iasql.com>
+- Alejandro Guillen <alejandro@iasql.com>
+- Yolanda Robla <yolanda@iasql.com>
+- Mohammad Pabandi <mohammad@iasql.com>
+
+### Implementation
+
+- [ ] Implemented: [One or more PRs](https://github.com/iasql/iasql-engine/some-pr-link-here) YYYY-MM-DD
+- [ ] Revoked/Superceded by: [RFC ###](./000 - RFC Template.md) YYYY-MM-DD
+
+## Author(s)
+
+- Yolanda Robla <yolanda@iasql.com>
+
+## Summary
+
+When performing deletion of resources, some of them cannot be deleted because they may be locked for some period of time. When a service is deleted, some of their dependencies may be locked for some period of
+time - even in the range of 24 hours - until the are totally released and can be removed from the cloud.
+This will cause some errors and inconsistencies when managing resources from our engine, and we need to have a way to manage these deletions asynchronously.
+
+## Proposal
+
+We could use the concept of `Garbage collection`, where we can schedule some of those resources for deletion at a certain point of time, and let those be collected regularly by a garbage collector, that will
+check if the resource is in a situation to be deleted and perform it.
+This can be achieved by using a table to track the resources marked for asynchronously deletion, and a periodicaly process that iterates over this table and triggers the proper deletion. Records of this table will contain:
+
+- module and entity of the resource to be deleted
+- cloudId for the resource
+- expected deletion date
```

Comment:
what do you mean with this?

Reply:
resources cannot be deleted until some expected date. Some of them in 1 hour, some of them in 24 hours,etc... so when we add the resource to the datable, we need to add a "deletion date" so the sync process can pick it when it's just on time, and avoid reading resources not ready to be deleted

Reply:
hmm I see your point is a good one. What I'm not sure about is how to make this generic but should be doable.